chatsky/utils/context_dict/ctx_dict.py

```diff
@@ -0,0 +1,203 @@
+from asyncio import gather
+from typing import Any, Awaitable, Callable, Dict, Generic, List, Mapping, Optional, Sequence, Set, Tuple, TypeVar, Union, Literal
+
+from pydantic import BaseModel, PrivateAttr, model_serializer, model_validator
+
+from chatsky.context_storages.database import DBContextStorage
+
+K, V = TypeVar("K"), TypeVar("V")
+
+
+async def launch_coroutines(coroutines: List[Awaitable], is_async: bool) -> List[Any]:
+    if is_async:
+        return await gather(*coroutines)
+    else:
+        return [await coroutine for coroutine in coroutines]
+
+
+class ContextDict(BaseModel, Generic[K, V]):
+    WRITE_KEY: Literal["WRITE"] = "WRITE"
+    DELETE_KEY: Literal["DELETE"] = "DELETE"
+
+    _items: Dict[K, V] = PrivateAttr(default_factory=dict)
+    _keys: List[K] = PrivateAttr(default_factory=list)
+    _hashes: Dict[K, int] = PrivateAttr(default_factory=dict)
+    _added: List[K] = PrivateAttr(default_factory=list)
+    _removed: List[K] = PrivateAttr(default_factory=list)
+
+    _storage: Optional[DBContextStorage] = PrivateAttr(None)
+    _ctx_id: str = PrivateAttr(default_factory=str)
+    _field_name: str = PrivateAttr(default_factory=str)
+    _field_constructor: Callable[[Dict[str, Any]], V] = PrivateAttr(default_factory=dict)
```

Comment:
I don't understand the purpose of this field.
AFAIK if `V` is a `BaseModel`, it will be validated.
And the constructor seems to only be called in `_load_items` and not in `connected`.

Reply:
OK, makes sense.
It still seems to me that `_field_constructor` is not called on values loaded in `connected`.
Or is `V` supposed to have a deserializer to init from bytes so that `cls.model_validate` calls `_field_constructor` automatically?